lib/modules/showImages.js

```diff
@@ -1910,9 +1790,19 @@ addModule('showImages', (module, moduleID) => {
 
 			return acceptRegex.test(href) && !rejectRegex.test(href);
 		},
-		handleLink(elem) {
+		async handleLink(elem) {
 			elem.type = 'IMAGE';
 			elem.src = elem.href;
+
+			if (module.options.convertGifstoGfycat.value && (/^(http|https|ftp):\/\/.*\.gif($|\/?)/).test(elem.href)) {
+				const { gfyName: id } = await RESEnvironment.ajax({
+					url: RESUtils.string.encode`//upload.gfycat.com/transcodeRelease?fetchUrl=${elem.href}`,
+					type: 'json',
+					cacheFor: RESUtils.DAY
+				});
+
+				if (id) await module.siteModules['gfycat'].handleLink(elem, [undefined, id]);
```

Comment:
I was mistaken about `convertGifToGfycat`, but found a way to keep it and avoid redundancy.